lib/os_mon/src/memsup.erl

```diff
@@ -843,6 +850,21 @@ clear_alarms() ->
 		  end,
 		  get()).
 
+maybe_check_process_mem(#state{ sys_only = false } = State, {Alloc, Total}) ->
+    {Pid, Bytes} = get_worst_memory_user(),
+    Threshold = State#state.proc_mem_watermark*Total,
+    if
+        Bytes > Threshold ->
+            set_alarm(process_memory_high_watermark, Pid);
+
+        true ->
+            clear_alarm(process_memory_high_watermark)
+    end,
+    State#state{mem_usage={Alloc, Total},
+                worst_mem_user={Pid, Bytes}};
+maybe_check_process_mem(State, _) ->
+    State.
```

Comment:
Maybe should set `mem_usage` as [before](https://github.com/emqx/otp/pull/21/files#diff-39f19df8821650991a8e7b166f5b8754b2103186f31d6c4b5eb9ec7de1207bc7L416)?

```suggestion
    State#state{mem_usage={Alloc, Total}}.
```

Reply:
Nice catch. 
I double checked that `mem_usage` is used by get_memory_data/1 which is different than `get_system_memory_data` so I have to update the `mem_usage` in the caller instead.